modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4020,6 +4024,17 @@ public void updateCodegenPropertyEnum(CodegenProperty var) {
         }
     }
 
+    private void updateEnumVarsWithExtensions(List<Map<String, Object>> enumVars, Map<String, Object> vendorExtensions) {
+        if (vendorExtensions != null && vendorExtensions.containsKey("x-enum-varnames")) {
+            List<String> alias = (List<String>) vendorExtensions.get("x-enum-varnames");
+            if (alias.size() == enumVars.size()) {
```

Comment:
Maye you could just log a warning if the sizes do not match.

In my opinion it is OK to still change the first two defined names in a case like this:

```yaml
      EnumProp:
        type: string
        enum:
          - a
          - b
          - c
        x-enum-varnames:
          - FOO
          - BAR
```

Reply:
I see.
Then I'll make changes to allow to set different number of elements from target enum.